leader-election-impl/src/main/java/com/palantir/paxos/PaxosQuorumChecker.java

```diff
@@ -189,90 +173,50 @@ private PaxosQuorumChecker() {
             }
         }
 
-        List<Throwable> toLog = Lists.newArrayList();
+        List<Throwable> encounteredErrors = Lists.newArrayList();
         boolean interrupted = false;
-        List<RESPONSE> receivedResponses = new ArrayList<RESPONSE>();
-        int acksRecieved = 0;
-        int nacksRecieved = 0;
-
+        PaxosResponses<RESPONSE> receivedResponses = new PaxosResponses<>(remotes.size(), quorumSize,
+                shortcircuitIfQuorumImpossible);
         try {
-            long deadline = System.nanoTime() + TimeUnit.SECONDS.toNanos(remoteRequestTimeoutInSec);
-            // handle responses
-            while (acksRecieved < quorumSize) {
+            long deadline = System.nanoTime() + remoteRequestTimeout.toNanos();
+            while (receivedResponses.shouldProcessNextRequest()) {
```

Comment:
Can we make this clearer via `hasNextRequest` and `shouldProcessNextRequest` perhaps? We're hoping to exit here because there are no remaining requests, not because we choose not to process one.